### ISSUE TITLE
Rule Change: Add `checkFootnoteDefinitions` option to `no-duplicate-definitions` and `no-unused-definitions` rules

### ISSUE THREAD

**Issue by ckhe1215** [eslint/markdown#611]
### What rule do you want to change?

`no-duplicate-definitions` and `no-unused-definitions`

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```markdown
Content[^footnote]

[^footnote]: This is a footnote
[^footnote]: This is a duplicate footnote

[^unused]: This footnote is never referenced
```

### What does the rule currently do for this code?

- `no-duplicate-definitions`: Reports duplicate `[^footnote]` definition with no way to disable footnote checking
- `no-unused-definitions`: Reports unused `[^unused]` definition with no way to disable footnote checking

### What will the rule do after it's changed?

When configured with `{ checkFootnoteDefinitions: false }`:

- `no-duplicate-definitions`: Will not report duplicate footnote definitions (while still checking regular definitions)
- `no-unused-definitions`: Will not report unused footnote definitions (while still checking regular definitions)

This change will provide consistency with `no-empty-definitions` rule and give users more granular control over footnote-related linting.

### Participation

- [x] I am willing to submit a pull request to implement this change.
